File: lua/grade-school/grade-school.lua
local School = {}
School.__index = School

function School:new()
  return setmetatable({ students = {} }, School)
end

function School:roster()
  return self.students
end

function School:add(name, grade)
  self.students[grade] = self.students[grade] or {}
  table.insert(self.students[grade], name)
  table.sort(self.students[grade])
end

function School:grade(gradeNumber)
  return self.students[gradeNumber] or {}
end

return School

File: php/grade-school/GradeSchool.php
<?php

declare(strict_types=1);

class School
{
    private $students = [];
    public function add(string $name, int $grade): void
    {
        $this->students[$grade][] = $name;
    }
    public function grade($grade)
    {
        return $this->students[$grade] ?? [];
    }
    public function studentsByGradeAlphabetical(): array
    {
        ksort($this->students);
        foreach (array_keys($this->students) as $grade) {
            sort($this->students[$grade]);
        }
        return $this->students;
    }
}

